pagopa/io-app
# chore: [IOBP-688] Adapt `ListItemTransaction` refactor and upgrade `io-app-design-system` to the version `3.0.1`

## Short description
This PR adapts the use of `ListItemTransaction`component with the breaking changes introduced in the `io-app-design-system` version `3.0.0`.

## List of changes proposed in this pull request
- Upgraded the `io-app-design-system` library to the latest version `3.0.1`;
- Adapted the changes to the `ListItemTransaction` usage;
- Edited the `getBadgeTextByTransactionStatus` function with the `getBadgeTextAndVariantByTransactionStatus`;

## How to test
Check that the following screens shows correctly the transaction item without any regression:
- Payments home screen;
- List of notices from the "Show all" CTA inside the Payments home screen;
- List of legacy notices from the "Mostra precedenti" CTA inside the list `PAYMENT_TRANSACTION_BIZ_EVENTS_LIST_SCREEN`;
- List of the transactions of a discount bonus with the IDPay feature flag enabled
    - To access this screen, after enabling IDPay FF -> Open the "Wallet tab" and choose a bonus initiative from the list;
